.gitignore

```diff
@@ -1 +1,66 @@
-*.swp
+#==============================================================================#
+# File extensions to be ignored anywhere in the tree.
+#==============================================================================#
+
+# Temp files created by most text editors
+*~
+
+# Merge files created by git
+*.orig
+
+# Byte compiled python modules
+*.pyc
+*.pyd
+
+# Vim swap files
+.*.sw?
+.sw?
+
+# Visual Studio
+.vs
+/.vscode/*
+
+# Sublime Text settings
+*.sublime-workspace
+*.sublime-project
+
+# Eclipse Project settings
+*.*project
+.settings
+
+# OS X specific files
+.DS_store
+
+#==============================================================================#
+# Explicit files to ignore (only matches one).
+#==============================================================================#
+
+# Various tags
+/tags
+/TAGS
+/GPATH
+/GRTAGS
+/GSYMS
+/GTAGS
+/ID
+.gitusers
+/compile_commands.json
+
+/CMakeSettings.json
+
+# Visual studio files
+!/.vscode/extensions.json
+!/.vscode/settings_recommended.json
```

Comment:
What does the `!` do here?

Reply:
! negates the next pattern. I think these lines would actually include these files explicitly. I'll remove them for now.